### PR TITLE
refactory(ts): almost all files are typescript now :rocket:

### PR DESCRIPTION
**Pull request by wwwzbwcom**
Create a pull request for easier checking changes first

### CODE REVIEW

File: src/modules/scf/index.ts [hunk 1]
```diff
@@ -1,39 +1,66 @@
...
 
-  async request({ Action, ...data }) {
-    const result = await Apis[Action](this.capi, data);
+  async request({ Action, ...data }: { Action: ActionType; [key: string]: any }) {
+    const result = await APIS[Action](this.capi, data);
     return result;
   }
 
-  // bind SCF_QcsRole role
+  // bind SCF_QcsRole roless
```
Comment (yugasun) [inline src/modules/scf/index.ts:63]: 单词错误，这里只是一个角色

**Review by yugasun — CHANGES_REQUESTED** [review]

File: src/modules/apigw/index.ts [hunk 2]
```diff
@@ -332,6 +391,26 @@ class Apigw {
...
+  }: {
+    serviceId: string;
+    subDomain: string;
+    inputs: ApigwBindCustomDomainInputs;
+  }): Promise<ApigwBindCustomDomainOutputs[]> {
+    const { customDomains, oldState = {} } = inputs;
+    if (!customDomains) return [];
+
+    // 1. unbind all custom domain
+    this.unbindCustomDomain(serviceId, oldState?.customDomains ?? []);
```
[the remark this answers is not in the document]
Comment (wwwzbwcom) [reply src/modules/apigw/index.ts:407]: 已修复

File: src/modules/apigw/index.ts [hunk 3]
```diff
@@ -483,8 +571,22 @@ class Apigw {
...
+
+    interface Detail {
+      InnerSubDomain: string;
+      InternalSubDomain: string;
+      OuterSubDomain: string;
+
+      ServiceId: string;
+
+      // FIXME: 小写？
+      serviceName: string;
```
[the remark this answers is not in the document]
Comment (wwwzbwcom) [reply src/modules/apigw/index.ts]: 已验证返回值均为 PascalCase

File: src/modules/apigw/index.ts [hunk 4]
```diff
@@ -583,19 +706,19 @@ class Apigw {
...
+    const output: ApiDeployerOutputs = {
       path: endpoint.path,
       method: endpoint.method,
       apiName: endpoint.apiName || 'index',
-      apiId: undefined,
       created: true,
       authType: authType,
       businessType: businessType,
       isBase64Encoded: endpoint.isBase64Encoded === true,
+      authRelationApiId: endpoint.authRelationApiId,
```
[the remark this answers is not in the document]
Comment (wwwzbwcom) [reply src/modules/apigw/index.ts:716]: 通过 deepClone 过滤输出中 undefined

File: src/modules/cls/index.ts [hunk 5]
```diff
@@ -1,22 +1,29 @@
...
+      secretId: credentials.SecretId!,
+      secretKey: credentials.SecretKey!,
       token: credentials.Token,
       debug: false,
       expire: expire,
     });
   }
 
-  async deployLogset(inputs) {
+  async deployLogset(inputs:ClsDeployLogsetInputs = {} as any) {
```
[the remark this answers is not in the document]
Comment (wwwzbwcom) [reply src/modules/cls/index.ts]: 已配置 typescript eslint + prettier

**Pull request merged**